docs/api-reference/cpp-api.rst

```diff
@@ -17,12 +17,136 @@ To search an API, refer to the API :ref:`genindex`.
 
 Device functions
 ================
+
+To use the device API, include the file ``rocrand_kernel.h`` in files that define kernels that use rocRAND device functions. The typical usage of device functions consists of the following operations in the device kernel definition:
+
+1. Create a new generator state object of the desired generator type.
+
+2. Initialize the generator state parameters using ``rocrand_init``.
+
+3. Generate random numbers by calling generation function on the generator state.
+
+4. Use the results.
+
+Since the rocRAND device functions are invoked from inside the user kernel, the generated numbers may be used right away in the kernel without a need to copy them to the host memory.
```

Comment:
```suggestion
Since the rocRAND device functions are invoked from inside the user kernel, the generated numbers can be used right away in the kernel without the need to copy them to the host memory.
```